### DISCUSSION TITLE
HxGrid - Restore to page X?

### DISCUSSION THREAD

**Discussion by nironixon** [havit/Havit.Blazor#306] [Q&A] [answered]
Hi there, 
I'm trying to find a way to restore a HxGrid to a certain page after a user has visited other pages in our application. 
Im saving the latest HxGrid property "StartIndex " with the ProtectedSessionStore provider but cant get my head around how to initialize the grid when the user sometime later visits my page with the grid. 

Question, when method OnInitializedAsync is running, is it possible to tell the HxGrid that I want it to be initialize on e.g. page "X"?

Regards Niclas

**Comment by hakenr** [comment, marked as answer]
Hi @nironixon,
use the `CurrentUserState` parameter (`@bind` it to your storage-backed field).

**Reply by nironixon** [reply]
Works like a charm!!! Thanks for an awesome lib!
